dartagnan/src/main/java/com/dat3m/dartagnan/program/event/lang/dat3m/DynamicThreadJoin.java

```diff
@@ -0,0 +1,89 @@
+package com.dat3m.dartagnan.program.event.lang.dat3m;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.ExpressionVisitor;
+import com.dat3m.dartagnan.expression.type.AggregateType;
+import com.dat3m.dartagnan.expression.type.IntegerType;
+import com.dat3m.dartagnan.expression.utils.IntegerHelper;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.AbstractEvent;
+import com.dat3m.dartagnan.program.event.BlockingEvent;
+import com.dat3m.dartagnan.program.event.RegReader;
+import com.dat3m.dartagnan.program.event.RegWriter;
+import com.google.common.base.Preconditions;
+
+import java.math.BigInteger;
+import java.util.HashSet;
+import java.util.Set;
+
+
+/*
+    This event joins with given thread, i.e., it blocks until the target thread terminates properly.
+    - The target thread is specified by a thread id, typically the one returned by DynamicThreadCreate.
+    - The type of the return register must be { bv? status, TRet retValue } where TRet may be void if no value is returned.
```

Comment:
what does `bv?` means here?

Reply:
It should indicate that different sizes are valid. The next lines explain that the type must be large enough to contain all error values. 
I can change it to `bvX` if you think it is clearer.